Makefile

```diff
@@ -38,6 +38,7 @@ TOP_DIR ?= $(UT_CORE_DIR)
 BIN_DIR ?= $(TOP_DIR)/build/bin
 LIB_DIR ?= $(TOP_DIR)/build/$(TARGET)/lib
 BUILD_DIR ?= $(TOP_DIR)/build/$(TARGET)/obj
+LIBRARY_DIR = $(LIB_DIR)
```

Comment:
remove this, it's outside of ut-core scope.

Reply:
`BUILD_WEAK_STUBS_SRC = xxx` is an optional flag, provided by the upper makefile, and send into this module.

In that case, we include the WEAK_STUBS as an independent library, and include them into the link list.

The `BUILD_WEAK_STUBS_SRC` if not define, will operate without any change

Reply:
```c
61: XLDFLAGS += $(YLDFLAGS) $(LDFLAGS) -L$(UT_CONTROL)/build/$(TARGET)/lib -lut_control -Wl,-rpath, -pthread -lpthread`
```

Needs to be upgraded to support the new weak_library, should be built into the `build/$(TARGET)/lib` directory.

```c
XLDFLAGS += -L$(BUILD_DIR)/$(TARGET)/lib -lweak_stubs_libs.a $(YLDFLAGS) $(LDFLAGS) -L$(UT_CONTROL)/build/$(TARGET)/lib -lut_control -Wl,-rpath, -pthread -lpthread 
````

Reply:
`YLDFLAGS` - is for the upper make file, state specific libraries and naming conventions that are defined in the specification for the module... Which is unknown to the ut-core makefile, but this is required to be as per the specification only a single .so which the vendor will implement.